frontend/public/components/dashboard/inventory-card/inventory-item.tsx

```diff
@@ -45,102 +54,161 @@ const getStatusGroupIcons = (flags: FlagsObject) => {
-    return (
-      <InventoryItem
-        isLoading={isLoading}
-        singularTitle={singularTitle}
-        pluralTitle={pluralTitle}
-        count={resources.length}
-        error={error}
-      >
-        {Object.keys(groups).filter(key => groups[key].count > 0).map(key => showLink ?
-          (
-            <StatusLink
-              key={key}
-              kind={kind}
-              namespace={namespace}
-              groupID={key}
-              count={groups[key].count}
-              statusIDs={groups[key].statusIDs}
-              filterType={groups[key].filterType}
-              flags={flags}
-            />
-          ) : (
-            <Status
-              groupID={key}
-              count={groups[key].count}
-              flags={flags}
-            />
-          )
-        )}
-      </InventoryItem>
-    );
-  }
-));
+)(({
+  kind,
+  useAbbr,
+  resources = [],
+  additionalResources,
+  isLoading,
+  mapper,
+  namespace,
+  error,
+  showLink = true,
+  flags = {},
+  ExpandedComponent,
+}) => {
+  const TitleComponent = React.useCallback((props) => (
+    <ResourceTitleComponent kind={kind} namespace={namespace} {...props} />
+  ), [kind, namespace]);
+
+  const groups = mapper(resources, additionalResources);
+  const top3Groups = getTop3Groups(Object.keys(groups).filter(key => groups[key].count > 0), flags);
+  return (
+    <InventoryItem
+      isLoading={isLoading}
+      title={useAbbr ? kind.abbr : kind.label}
+      count={resources.length}
+      error={error}
+      TitleComponent={showLink ? TitleComponent : null}
+      ExpandedComponent={ExpandedComponent}
+    >
+      {top3Groups.map(key => showLink ?
+        (
+          <StatusLink
+            key={key}
+            kind={kind}
+            namespace={namespace}
+            groupID={key}
+            count={groups[key].count}
+            statusIDs={groups[key].statusIDs}
+            filterType={groups[key].filterType}
+          />
+        ) : (
+          <Status
+            key={key}
+            groupID={key}
+            count={groups[key].count}
+          />
+        )
+      )}
+    </InventoryItem>
+  );
+});
 
-export type StatusGroupMapper = (resources: K8sResourceKind[], additionalResources?: {[key: string]: K8sResourceKind[]}) => {[key in InventoryStatusGroup | string]: {filterType?: string, statusIDs: string[], count: number}};
+export type StatusGroupMapper = (
+  resources: K8sResourceKind[],
+  additionalResources?: {[key: string]: K8sResourceKind[]},
+) => {[key in InventoryStatusGroup | string]: {filterType?: string, statusIDs: string[], count: number}};
 
 type InventoryItemProps = {
   isLoading: boolean;
-  singularTitle: string;
-  pluralTitle: string;
+  title: string;
   count: number;
   children?: React.ReactNode;
   error: boolean;
+  TitleComponent?: React.ComponentType<any>;
+  ExpandedComponent?: React.ComponentType<any>;
```

Comment:
Nit: `ComponentType<P = {}>` has fallback to `{}` so do we really need an explicit `any` type parameter?

I think that `{}` is a better (more sensible) type fallback than `any` when it comes to React component props.

Reply:
ok, i wasnt sure which is better..using `ComponentType<{}>`